src/expycted/matchers/equal_matcher.py

```diff
@@ -11,4 +12,13 @@ class EqualMatcher(BaseMatcher):
     OPERATION = "=="
 
     def _matches(self, *, expected: Any) -> bool:
-        return self._expectation.actual == expected
+        if not self._expectation.qualifiers.weak:
+            return self._expectation.actual == expected
+
+        return any(
+            [
+                str(self._expectation.actual) == str(expected),
+                pickle.dumps(self._expectation.actual) == pickle.dumps(expected),
+                self._expectation.actual == expected,
+            ]
+        )
```

Comment:
Please refactor to *lazy* loaded callbacks to allow for a performance increase.

Also, specific tests triggering the `str` and `pickle` functionality should be added.